beluga_system_tests/test/test_system_new.cpp

```diff
@@ -141,24 +141,18 @@ auto particle_filter_test(
   auto probability_estimator = beluga::ThrunRecoveryProbabilityEstimator(params.alpha_slow, params.alpha_fast);
 
   // Iteratively run the filter through all the data points.
+  beluga::RollingWindow<Sophus::SE2d, 2> control_action_window{};
   for (auto [measurement, odom, ground_truth] : datapoints) {
     if (!should_update(odom)) {
       continue;
     }
 
     // TODO(nahuel): Update this once the new model features are ready.
     /**
-     * particles |= beluga::actions::propagate(std::execution::par, motion(odom)) |
-     *              beluga::actions::reweight(std::execution::par, sensor(measurement));
+     * particles |= beluga::actions::reweight(std::execution::par, sensor_model(measurement));
      */
-    motion.update_motion(odom);
     sensor.update_sensor(std::move(measurement));
-    particles |= beluga::actions::propagate(
-                     std::execution::par,
-                     [&motion](const auto& state) {
-                       static thread_local auto engine = std::mt19937{std::random_device()()};
-                       return motion.apply_motion(state, engine);
-                     }) |
+    particles |= beluga::actions::propagate(std::execution::par, motion(control_action_window << odom)) |
```

Comment:
This is the best I could come up with that is still a one-liner. It is in line with what `Eigen` does to populate matrices, but I know it may be somewhat controversial.

Reply:
Looks great to me!